packages/visualizations/src/components/Table/Row.svelte

```diff
@@ -0,0 +1,105 @@
+<script lang="ts">
+    import Cell from './Cell';
+    import ZoomIcon from './ZoomIcon.svelte';
+    import type { Column, RowProps } from './types';
+
+    export let columns: Column[];
+    export let rowProps: RowProps | undefined;
+    export let record: Record<string, unknown>;
+
+    let isRowHovered = false;
+    $: ({ onClick, onMouseEnter, onMouseLeave, actionAriaLabel } = rowProps || {});
+    $: handleMouseEnter = () => {
+        if (onMouseEnter) {
+            onMouseEnter(record);
+        }
+        isRowHovered = true;
+    };
+    $: handleMouseLeave = () => {
+        if (onMouseLeave) {
+            onMouseLeave(record);
+        }
+        isRowHovered = false;
+    };
+    $: handleClick = () => {
+        if (onClick) {
+            onClick(record);
+        }
+    };
+</script>
+
+<tr
+    on:mouseenter={rowProps && handleMouseEnter}
+    on:mouseleave={rowProps && handleMouseLeave}
+    on:focusin={rowProps && handleMouseEnter}
+    on:focusout={rowProps && handleMouseLeave}
+>
+    {#if rowProps?.onClick}
+        <td class="button-cell">
+            <button
+                on:click={rowProps && handleClick}
+                aria-label={actionAriaLabel || 'Expand Record'}
+            >
+                <span class:visually-hidden={!isRowHovered}>
```

Comment:
The mouseenter/mouseleave can be tricky and skipped in some cases. In our use case with a modal I think mouseleave is never triggered.

https://github.com/user-attachments/assets/1197ab1d-287e-4719-8594-ab1266c1a9f5

Maybe this can be achieved instead by a regular `tr:hover ...` in CSS to visually hide the button? (which as far as I know handles that case well)

Reply:
ha, I get it it's the focusin/out that does that. It worked without it but then we had no keyboard and now I brought it back we don't lose focus when opening the modal. I'll check if it's fixable only on the platform side.